### PR TITLE
Add .flutterflowignore support

### PR DESCRIPTION
**Pull request by tmpsantos**
This file allows for users to create a list of files to be ignored when exporting the project code (works a bit like .gitignore).

Files matching the globbing rules listed at .flutterflowignore are going to be ignored. This might break the project if changes in these files are needed to compile the project, so it should be used with extreme caution.

This functionality was previously achieved with:
https://github.com/krabhishek/flutterflow-filtered-pull

### CODE REVIEW

**Comment by tmpsantos** [comment]
cc @aemelyanovff

**Comment by aemelyanovff** [comment]
> cc @aemelyanovff

I'll review it hopefully by the end of the week. I'm a bit swamped right now, sorry.

File: lib/src/flutterflow_ignore.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  bool matches(String path) {
+    for (final glob in _globs) {
+      if (glob.matches(path)) {
+        return true;
+      }
+    }
+
+    return false;
```
Comment (aemelyanovff) [inline lib/src/flutterflow_ignore.dart:41]: Nit: `return _globs.any((glob) => glob.matches(path));` is a little more idiomatic in Dart.

File: lib/src/flutterflow_api_client.dart [hunk 2]
```diff
@@ -135,25 +133,28 @@ Future<String?> exportCode({
...
   for (final file in projectFolder.files) {
     if (file.isFile) {
-      final data = file.content as List<int>;
-      final filename = file.name;
+      final relativeFilename =
+          path_util.joinAll(path_util.split(file.name).sublist(1));
+
+      // Found on .flutterflowignore, move on.
+      if (ignore.matches(relativeFilename)) {
+        print('Ignoring $relativeFilename, file remained unchanged.');
```
Comment (aemelyanovff) [inline lib/src/flutterflow_api_client.dart:147]: Nit: logs like this usually go to `stderr` (it's debatable of course).

File: test/fixtures/.flutterflowignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+file
+dir/*
```
Comment (aemelyanovff) [inline test/fixtures/.flutterflowignore:2]: When I add `dir/` to `.gitignore` it ignores the entire directory, but this implementation doesn't. Let's update it to support ignoring directories?

**Review by aemelyanovff — COMMENTED** [review]
Very nice!
I have one suggestion and a couple of nits.

File: test/fixtures/.flutterflowignore (hunk 3 above)
Comment (tmpsantos) [reply test/fixtures/.flutterflowignore:2]: Done, also added support for comments starting with `#` like in git.

File: lib/src/flutterflow_ignore.dart (hunk 1 above)
Comment (tmpsantos) [reply lib/src/flutterflow_ignore.dart]: Done.

File: lib/src/flutterflow_api_client.dart (hunk 2 above)
Comment (tmpsantos) [reply lib/src/flutterflow_api_client.dart]: Makes sense, the rest of the code is like this.

**Comment by tmpsantos** [comment]
Thanks @aemelyanovff for the review. Addressed your concerns and added more tests and comments support. Also made the path being match always relative to where the `.flutterflowignore` is located for the sake of consistence.

**Comment by aemelyanovff** [comment]
LGTM!

**Pull request merged**